src/register/RegistrationFields/CountryField/CountryField.jsx

```diff
@@ -65,30 +71,27 @@ const CountryField = (props) => {
 
     const { value } = event.target;
 
-    const { countryCode, displayValue, error } = validateCountryField(
+    const { error } = validateCountryField(
       value.trim(), countryList, formatMessage(messages['empty.country.field.error']), formatMessage(messages['invalid.country.field.error']),
     );
-
-    onChangeHandler({ target: { name: 'country' } }, { countryCode, displayValue });
     handleErrorChange('country', error);
   };
 
-  const handleSelected = (value) => {
-    handleOnBlur({ target: { name: 'country', value } });
-  };
-
   const handleOnFocus = (event) => {
     handleErrorChange('country', '');
     dispatch(clearRegistrationBackendError('country'));
     onFocusHandler(event);
   };
 
   const handleOnChange = (value) => {
-    onChangeHandler({ target: { name: 'country' } }, { countryCode: '', displayValue: value });
+    onChangeHandler({ target: { name: 'country' } }, { countryCode: value.selectionId, displayValue: value.userProvidedText });
```

Comment:
```suggestion
    onChangeHandler({ target: { name: 'country' } }, { countryCode: value.selectionValue, displayValue: value.userProvidedText });
```

Reply:
This selectionValue does not have country code it keep display value which is normal country name not its code that is why I have to use sectionId instead of selectionValue

Reply:
then for consistency we should also change this (set displayValue in selectionValue)

```

  const countryFieldValue = {
    userProvidedText: selectedCountry.displayValue,
    selectionValue: selectedCountry.displayValue,
    selectionId: selectedCountry.countryCode,
  };
```